Library/Homebrew/cmd/desc.rb

```diff
@@ -15,18 +21,24 @@ module Homebrew
 
   def desc
     search_type = []
-    search_type << :either if ARGV.flag? "--search"
-    search_type << :name   if ARGV.flag? "--name"
-    search_type << :desc   if ARGV.flag? "--description"
+    search_type << :either    if ARGV.flag? "--search"
+    search_type << :name      if ARGV.flag? "--name"
+    search_type << :desc      if ARGV.flag? "--description"
+    search_type << :all       if ARGV.flag? "--all"
+    search_type << :installed if ARGV.flag? "--installed"
 
-    if search_type.empty?
+    if search_type.size > 1
+      odie "Pick one, and only one of -s/--search, -n/--name, or -d/--description -i/--installed -a/--all."
```

Comment:
You can avoid the `elsif` here as `odie` will `exit` immediately.  Additionally, I find this "pick one" pretty hard to parse, personally.